### PR TITLE
Subscribe/Unsubscribe Logic

### PR DESCRIPTION
**Pull request by derekdowling**
Provides component/action/store logic for following/unfollowing people. Also created an Author object with help functions. Yay!

### CODE REVIEW

**Comment by derekdowling** [comment]
@KarimBaaba ready for review!

**Comment by derekdowling** [comment]
Contributes to #23 and closes #15 still need to add ajax.

File: app/scripts/components/follow.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        break;
+      case 'Follow':
+      case 'Friend':
+        AuthorActions.subscribeTo(this.props.author);
+        break;
+    }
+
+    // hack to rerender follow button rather than making the whole app rerender
+    // because the current user updates
+    setTimeout(function () {this.forceUpdate()}.bind(this), 50);
```
Comment (xkb) [inline app/scripts/components/follow.jsx:23]: Wouldn't be better if this component subscribed to AuthorStore 's "friends list"?

File: app/scripts/objects/author.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  findAuthorById (authors, id) {
+    for (let author of authors) {
+      if (author.id == id) {
+        return true;
+      }
+    }
+    return false;
+  }
+}
```
Comment (xkb) [inline app/scripts/objects/author.js:99]: Fuck Yeah.

File: app/scripts/components/follow.jsx (hunk 1 above)
Comment (derekdowling) [inline app/scripts/components/follow.jsx:23]: Perhaps. I'm not sure the best way to do this without a big refactor to the AuthorStore. I've tried to compartmentalize all of the subscription logic.

Comment (xkb) [inline app/scripts/components/follow.jsx:23]: You can add properties to the author that expose `hasFriend`, `follows` and `followedBy` as list objects that this component can subscribe to using `listenTo` or `connect`. Does that make sense?

File: gulpfile.js [hunk 3]
```diff
@@ -20,8 +20,8 @@ gulp.task('clean:dist', function() {
 gulp.task('scripts', function() {
   var bundler = browserify('./app/scripts/app.js', {
     debug: true,
-    extensions: ['.es6','.jsx']
-  }).transform(to5ify);
+    extensions: ['.js','.jsx']
+  }).transform(babelify);
 
   return bundler.bundle()
     .pipe(source('app.js'))
```
Comment (xkb) [inline gulpfile.js:27]: Thanks, I was getting around to do this.

File: app/scripts/components/follow.jsx (hunk 1 above)
Comment (derekdowling) [inline app/scripts/components/follow.jsx:23]: Okay, yeah, I can probably do that.

File: app/index.html [hunk 4]
```diff
@@ -11,14 +11,12 @@
...
+    <link rel="stylesheet" href="font-awesome/css/font-awesome.min.css">
     <link rel="stylesheet" href="styles/main.css">
     <!-- endbuild -->
 
     <!-- build:js vendor/modernizr.js -->
     <script src="modernizr/modernizr.js"></script>
-    <script src="jquery/dist/jquery.min.js"></script>
-    <script src="bootstrap/dist/js/bootstrap.min.js"></script>
-    <script src="alertifyjs/dist/js/alertify.js"></script>
     <!-- endbuild -->
```
Comment (xkb) [inline app/index.html:20]: Did this break something?

**Pull request merged**